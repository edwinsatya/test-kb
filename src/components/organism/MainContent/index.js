import React from "react";
import Card from "../../molecules/Card";
import { useSelector } from "react-redux";

export default function MainContent() {
  const dataApi = useSelector((state) => state.dataApi);
  const errorMsg = useSelector((state) => state.error);
  const loading = useSelector((state) => state.loading);

  if (loading) {
    return <div>Loading..</div>;
  } else if (errorMsg) {
    return (
      <div className="mt-5 md:mt-8 xl:mt-16">
        <div className=" bg-red-300 p-5">
          <h2 className="text-lg font-bold">
            Response from server : {errorMsg}
          </h2>
        </div>
      </div>
    );
  } else {
    return (
      <div className="mt-5 md:mt-8 xl:mt-16 grid grid-cols-3 gap-5 md:gap-8 xl:gap-10">
        {dataApi.length > 0 ? (
          dataApi?.map((data, index) => (
            <Card
              key={index}
              classProps={
                "border border-gray-300 rounded-lg overflow-hidden shadow-xl transition-all duration-300 hover:scale-105 hover:shadow-none"
              }
              dataProps={data}
            ></Card>
          ))
        ) : (
          <div className="text-md font-bold">No Have Item</div>
        )}
      </div>
    );
  }
}
