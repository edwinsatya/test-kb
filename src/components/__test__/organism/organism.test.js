import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../../config/redux/store";
import MainContent from "../../organism/MainContent";

describe("MainContent Component", () => {
  it("Should render with store", () => {
    let component = renderer.create(
      <Provider store={store}>
        <MainContent />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
