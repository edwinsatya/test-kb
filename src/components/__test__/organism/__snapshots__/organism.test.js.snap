// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent Component Should render with store 1`] = `
<div
  className="mt-5 md:mt-8 xl:mt-16 grid grid-cols-3 gap-5 md:gap-8 xl:gap-10"
>
  <div
    className="text-md font-bold"
  >
    No Have Item
  </div>
</div>
`;
